app/Http/Controllers/Teacher/Dashboard.php

```diff
@@ -5,11 +5,20 @@
 namespace App\Http\Controllers\Teacher;
 
 use App\Http\Controllers\Controller;
+use Illuminate\Support\Facades\Auth;
+use App\Student;
 
 class Dashboard extends Controller
 {
     public function __invoke()
     {
-        return view('pages.teacher.dashboard');
+        $user = Auth::user();
+        $students = student::join('workplacelearningperiod', 'student.student_id', '=', 'workplacelearningperiod.student_id')
```

Comment:
This should be moved to a one-to-many relationship of Teacher (Student model) -> WorkplaceLearningPeriod relationship. This would allow us to reuse the code so we don't have to write the same query everytime we want to access the Teacher's students.

You can take inspiration from the current relationship of Student -> WorkplaceLearningPeriod(s):
[WorkplaceLearningPeriod](https://github.com/HUInstituteForICT/workplacelearning/blob/5aab6978a68e199f64ba5b966baf5b0dcd5d7610/app/WorkplaceLearningPeriod.php#L94-L97)
[Student](https://github.com/HUInstituteForICT/workplacelearning/blob/5aab6978a68e199f64ba5b966baf5b0dcd5d7610/app/Student.php#L185-L188)

Don't forget to check if Student model is actually a Teacher, we don't want to suddenly have actual Students attached to another Student's WorkplaceLearningPeriod.

Reply:
[Laravel documentation on relationships](https://laravel.com/docs/5.7/eloquent-relationships#one-to-many)